topic-controller/src/main/java/io/strimzi/controller/topic/Controller.java

```diff
@@ -621,53 +621,38 @@ void onTopicPartitionsChanged(TopicName topicName, Handler<AsyncResult<Void>> re
 
-                            // getting topic metadata from Kafka
-                            Topic kafkaTopic = TopicSerialization.fromTopicMetadata(metadataResult.result());
-
-                            // getting topic information from the private store
-                            topicStore.read(topicName, topicResult -> {
+                                // getting topic metadata from Kafka
+                                Topic kafkaTopic = TopicSerialization.fromTopicMetadata(metadataResult.result());
 
                                 // if partitions aren't changed on Kafka yet, we retry with exponential backoff
                                 if (topicResult.result().getNumPartitions() == kafkaTopic.getNumPartitions()) {
-
-                                    long delay;
-                                    try {
-                                        delay = backOff.delayMs();
-                                        logger.debug("Topic {} partitions changed, but metadata not updated in Kafka yet: Backing off for {}ms", topicName, delay);
-                                    } catch (MaxAttemptsExceededException e) {
-                                        logger.info("Topic {} partitions changed, but metadata not updated in Kafka after {}ms, giving up for now", topicName, backOff.totalDelayMs());
-                                        fut.fail(e);
-                                        return;
-                                    }
-
-                                    if (delay < 1) {
-                                        // vertx won't tolerate a zero delay
-                                        vertx.runOnContext(timerId -> kafka.topicMetadata(topicName, this));
-                                    } else {
-                                        vertx.setTimer(TimeUnit.MILLISECONDS.convert(delay, TimeUnit.MILLISECONDS),
-                                                timerId -> kafka.topicMetadata(topicName, this));
-                                    }
+                                    retry();
                                 } else {
                                     logger.info("Topic {} partitions changed to {}", topicName, kafkaTopic.getNumPartitions());
                                     Controller.this.reconcileOnTopicChange(topicName, kafkaTopic, fut.completer());
                                 }
-                            });
 
-                        } else {
-                            fut.fail(metadataResult.cause());
+                            } else {
+                                fut.fail(metadataResult.cause());
+                            }
                         }
-                    }
-                };
-                kafka.topicMetadata(topicName, handler);
+
+                        @Override
+                        public void onMaxAttemptsExceeded(MaxAttemptsExceededException e) {
+                            fut.complete();
```

Comment:
We should comment about why we complete here, rather than fail (because it might not be obvious to someone reading the code why it is OK to "give up" in this case)

Reply:
Done.